common/src/lib.rs

```diff
@@ -187,22 +187,38 @@ pub trait DAGBasedLedger {
 pub struct Dispatch {
     pub kind: DispatchKind,
     pub message: Message,
+    pub payload_store: Option<PayloadStore>,
 }
 
 impl Dispatch {
     pub fn new_init(message: Message) -> Self {
         let kind = DispatchKind::Init;
-        Dispatch { message, kind }
+        let payload_store: Option<PayloadStore> = None;
+        Dispatch {
+            message,
+            kind,
+            payload_store,
```

Comment:
Or even (as you do in some places):
```suggestion
            payload_store: None,
```